charts/opentelemetry-collector/values.yaml

```diff
@@ -279,8 +279,10 @@ lifecycleHooks: {}
 #       - "5"
 
 service:
+  enabledIfDaemonset: false
```

Comment:
Maybe instead of labeling this with `daemonset` we can use something more generic like `explicitlyEnable`?  That way if something else in the future wants to enable the service it will be able to.  You can also then remove the check for `mode = daemonset` in the template's `if` statement.

Reply:
Can't we use `internalTrafficPolicy==Local && mode=daemonset` as the condition for now? If we will need someone to have a regular service with a daemonset in future, we can always introduce this option that would take precedence.

Reply:
Yes that'll work